apps/aechannel/test/aesc_txs_SUITE.erl

```diff
@@ -2875,6 +2879,90 @@ fp_too_soon(Cfg) ->
     Test(100, 200, 100),
     ok.
 
+fp_register_name(Cfg) ->
+    Name = <<"lorem.test">>,
+    Salt = 42,
+    {ok, NameAscii} = aens_utils:to_ascii(Name),
+    CHash           = aeu_hex:hexstring_encode(
+                        aens_hash:commitment_hash(NameAscii, Salt)),
+    #{ public := Pubkey, secret := Privkey } = enacl:sign_keypair(),
+    Account = aeu_hex:hexstring_encode(Pubkey),
+    StateHashSize = aec_base58c:byte_size_for_type(state),
+    StateHash = <<42:StateHashSize/unit:8>>,
+    Round = 42,
+    FPRound = Round + 10,
+    Test =
+        fun(Owner, Forcer) ->
+            ContractCreateRound = 10,
+            run(#{cfg => Cfg},
+               [positive(fun create_channel_/2),
+                % store state on-chain via snapshot
+                set_from(initiator),
+                set_prop(round, 42),
+                set_prop(state_hash, StateHash),
+                positive(fun snapshot_solo_/2),
+                fun(#{state := S,
+                      channel_pubkey := ChannelPubKey} = Props) ->
+                    Channel = aesc_test_utils:get_channel(ChannelPubKey, S),
+                    Round = aesc_channels:round(Channel),
+                    StateHash = aesc_channels:state_hash(Channel),
+                    Props
+                end,
+                % create contract off-chain
+                create_trees_if_not_present(),
+                set_from(Owner, owner, owner_privkey),
+                create_contract_in_trees(_Round    = ContractCreateRound,
+                                         _Contract = "aens",
+                                         _InitArgs = <<"()">>,
+                                         _Deposit  = 2),
+                % force progress contract on-chain
+                fun(#{contract_id := ContractId} = Props) ->
+                    BinToSign = <<Pubkey/binary, ContractId/binary>>,
+                    <<Word1:256, Word2:256>> = enacl:sign_detached(BinToSign,
+                                                                   Privkey),
+                    Sig = aeu_hex:hexstring_encode(
+                            aeso_data:to_binary({Word1, Word2})),
+                    PreclaimArgs = <<"(", Account/binary, ",",
+                                          CHash/binary, ",",
+                                          Sig/binary,
+                                      ")">>,
+                    (force_call_contract_first(Forcer, <<"preclaim">>,
```

Comment:
I see this calls `CallData = aect_sophia:create_call(Code, Fun, Args)` without checking whether return value is `{error, _}` or binary (binary return is not tagged with `{ok, _}`).  If it returned the `{error, _}` then execution would fall at https://github.com/aeternity/epoch/blob/e6940492d22ea11e79df55575f3552e86a47752f/apps/aevm/src/aevm_eeevm_state.erl#L166 that is the first point that checks that it is binary, and sets gas to zero, that - if happenend - I am concerned may make this test test something else than intended, without failing (did not double check). Would you please consider making `CallData = aect_sophia:create_call(Code, Fun, Args)` more strict by at least failing if not a binary? (I would prefer a tagged `{ok, CallData}` return though not sure why not tagged with `{ok, _}` already.) It would at least improve readability of the test (that is already complex).